test/prif_init_test.F90

```diff
@@ -0,0 +1,67 @@
+module prif_init_test_m
+    use prif, only : prif_init, PRIF_STAT_ALREADY_INIT
+    use julienne_m, only: test_description_t, test_diagnosis_t, test_result_t, test_t, operator(.equalsExpected.), usher
+
+    implicit none
+    private
+    public :: prif_init_test_t
+
+    type, extends(test_t) :: prif_init_test_t
+    contains
+      procedure, nopass, non_overridable :: subject
+      procedure, nopass, non_overridable :: results
+    end type
+
+contains
+
+  pure function subject() result(test_subject)
+    character(len=:), allocatable :: test_subject
+    test_subject = "prif_init"
+  end function
+
+  function results() result(test_results)
+    type(test_result_t), allocatable ::  test_results(:)
+    type(prif_init_test_t) prif_init_test
+
+    test_results = prif_init_test%run([ &
+        test_description_t("completing successfully", usher(check_caffeination)) &
+       ,test_description_t("returning PRIF_STAT_ALREADY_INIT on a subsequent call ", usher(check_subsequent_prif_init_call)) &
+    ])
+  end function
+
+
+  function check_caffeination() result(diag)
+    ! this test needs to run very early at startup, so we memoize the result
+    type(test_diagnosis_t) :: diag
+    type(test_diagnosis_t), save :: memo
+    logical, save :: first_pass = .true.
+
+    if (first_pass) then
+      first_pass = .false.
+      write_memo: &
+      block
+#if HAVE_MULTI_IMAGE
+        integer, parameter :: successful_initiation = PRIF_STAT_ALREADY_INIT
+#else
+        integer, parameter :: successful_initiation = 0
+#endif
+        integer init_exit_code
+
+        call prif_init(init_exit_code)
+        memo = init_exit_code .equalsExpected. successful_initiation
+      end block write_memo
+    endif 
+
+    diag = memo
+  end function
```

Comment:
The comment and memoization in this function were presumably copied from test/a00_caffeinate_test.F90, but the corresponding early call to `check_caffeination` was not deployed in julienne-driver.F90 (see [main.F90](https://github.com/BerkeleyLab/caffeine/blob/0.6.0/test/main.F90#L79-L80)).

Does Julienne guarantee that the `test_fixture_t` elements in a `test_harness_t` are executed strictly in-order? 
* If so, then we should remove the comment and superfluous memoization from this function.
* If not, then we should explicitly call `check_caffeination` early in julienne-driver.F90

Reply:
@bonachea yes, Julienne executes the tests in the `test_harness_t` array in element order.  Please suggest the change and I'll accept it or feel free to suggest and accept it if that's possible.

Reply:
Resolved in b2977a62